### DISCUSSION TITLE
Using FreeCAD may reduce some codes and dependency

### DISCUSSION THREAD

**Discussion by ebrahimraeyat** [xcfem/xc#65] [Ideas]
Hi @lcpt, I think if we use FreeCAD (or better FreeCAD FEM Workbench), we can reduce dependency of xc like CGAL, VTK and reduce some codes that related to Solid (line, face, solid) object like centerofmass, xmax, Ix, Iy and etc). I think FreeCAD has many capabilities add-on for xc.
thanks, Ebrahim

**Comment by lcpt** [comment]
Hi, @ebrahimraeyat.

XC makes a heavy use of CGAL, not only in the preprocessor routines but also inside the material models to create interaction diagrams, compute intersections, etc., etc. 

I think that XC **needs** FreeCAD but not to replace CGAL or VTK but for making easier to create/modify complex models and as an interface with IFC. 

As you can see in the previous image, meshing works reasonably well. It's the creation of the model geometry (and assigning materials, thicknesses, loaded zones, etc.) which is really painful. I think that using FreeCAD models decorated with the appropriate metadata (materials, steel shapes, thicknesses, ...) is the solution. Moreover, if we can use the IFC mechanisms already in FreeCAD to manage that it will be great (data exchange, ...).

![connection_meshing_detail](https://user-images.githubusercontent.com/12873038/102633538-dc0d9980-4150-11eb-83e4-ddfe719e271e.png)

Thanks to you. Luis

**Reply by ebrahimraeyat** [reply]
Thanks @lcpt  for your complete explanation. If you have some roadmap for what it needed to implement in FreeCAD, maybe i can work on it. thanks.

**Reply by lcpt** [reply]
Thanks, @ebrahimraeyat 

I'm trying to figure out how to organize the data transfer between FreeCAD and XC and I wonder how you can assign metadata (for example the material name or the thickness) to the FreeCAD objects.

For example, I have this bridge deck model made with surfaces:

![bridge_deck](https://user-images.githubusercontent.com/12873038/102876245-36ef0b80-4445-11eb-8fce-7ea0ed59fd89.png)

Is there a way to assign a material and a thickness to each surface using the FEM workbench or the BIM module?

Thanks.

**Reply by ebrahimraeyat** [reply]
As i know, in FEM you can use shell element to assign thickness.

![image](https://user-images.githubusercontent.com/8196112/102882094-1cc42600-4463-11eb-825b-8ad2f92e239c.png)

![image](https://user-images.githubusercontent.com/8196112/102882174-3a918b00-4463-11eb-84b3-a6ee624e9715.png)

**Reply by ebrahimraeyat** [reply]
also the geometricall properties can assign with various tools in FreeCAD, not assign, drawing! 
like extrude in BIM or Part WB:

![image](https://user-images.githubusercontent.com/8196112/102882732-16827980-4464-11eb-938a-52c49fe394cd.png)

**Comment by ebrahimraeyat** [comment]
I modeled a base plate with solid and plate mesh:

[](https://forum.freecadweb.org/viewtopic.php?f=18&t=42390&hilit=base+plate)

![1](https://user-images.githubusercontent.com/8196112/102884846-95c57c80-4467-11eb-912d-eaee80d85b16.jpeg)

![2](https://user-images.githubusercontent.com/8196112/102884859-9bbb5d80-4467-11eb-8981-58019f16fa52.jpeg)

**Comment by lcpt** [comment]
Thank you! I'll take a look.
